apps/web/src/pages/templates/filter/FilterModal.tsx

```diff
@@ -19,11 +20,18 @@ export function FilterModal({
 }) {
   const theme = useMantineTheme();
 
-  const { fields, append, remove } = useFieldArray({
+  const { fields, append, update, remove } = useFieldArray({
     control,
     name: `steps.${stepIndex}.filters.0.children`,
   });
 
+  function handleOnChildOnChange(index: number) {
+    return (data) => {
+      const newField = Object.assign({}, fields[index], { on: data });
+      update(index, newField);
```

Comment:
Created this update in order to update the fields so the component rerenders, not so sure if it is the best to do so.